skrub/_table_vectorizer.py

```diff
@@ -458,6 +471,105 @@ def _clone_transformers(self):
 
         # TODO: check that the provided transformers are valid
 
+    def _split_univariate_transformers(self, split_fitted: bool = False):
+        """
+        Split univariate transformers into multiple transformers, one for each
+        column. This is useful to use the inherited `ColumnTransformer` class
+        parallelism.
+
+        Parameters
+        ----------
+        split_fitted : bool, default=False
+            Whether to split the self.transformers_ attribute (True) or the
+            self.transformers attribute (False). The former is used when
+            calling `transform` and the latter when calling `fit_transform`.
+        """
+        if split_fitted:
+            check_is_fitted(self, attributes=["transformers_"])
+            self._transformers_fitted_original = self.transformers_
+            new_transformers_ = []
+            new_transformer_to_input_indices = {}
+            for name, trans, cols in self.transformers_:
+                if (
+                    trans.__class__.__name__ in UNIVARIATE_TRANSFORMERS
+                    and len(cols) > 1
+                ):
+                    splitted_transformers_ = trans._split()
+                    assert len(splitted_transformers_) == len(cols)
+                    for i, col in enumerate(cols):
+                        new_transformers_.append(
+                            (
+                                name + "_split_" + str(i),
+                                splitted_transformers_[i],
+                                [col],
+                            )
+                        )
+                        new_transformer_to_input_indices[name + "_split_" + str(i)] = [
+                            self._transformer_to_input_indices[name][i]
+                        ]
```

Comment:
Nitpick (this also has the benefit of checking for sizes between the 3 lists)
```suggestion
                    for i, (col, trans, trans_to_mapping) in enumerate(zip(
                        cols,
                        splitted_transformers_,
                        self._transformer_to_input_indices[name]
                    ):
                        name_split = f"{name}_split_{i}"
                        new_transformers_.append(
                            (name_split, trans, [col])
                        )
                        new_transformer_to_input_indices[name_split] = [trans_to_mapping]
```